src/pybullet_industrial/g_code_processor.py

```diff
@@ -0,0 +1,435 @@
+        """Reads g-code row by row and saves the processed data in
+        a list.
+        Comments that start with % are ignored and all the other data is
+        stored as it gets read in.
+
+        Args:
+            filename (list[str]): Source of information
+
+        Returns:
+            gcode (list)
+        """
+
+        gcode_input = gcode_input.splitlines()
+        gcode = []
+
+        # Loop over the lines of the file
+        for line in gcode_input:
+
+            if not line.strip():
+                continue
+
+            # Read in G-Code if line is not a comment and not empty
+            if line[0] != "%" and len(line) > 1:
+
+                # Initialize a new line as a list
+                new_line = []
+
+                # Split the line into its components
+                data = line.split()
+
+                # Loop over the components
+                for i in data:
+                    # Determine the ID of the component
+                    id_val = i[0]
+
+                    # Extract the value of the component
+                    val2 = float(i[1:])
+
+                    if id_val in ["G", "M", "T"]:
+                        # Insert the value into the corresponding
+                        # column of the new line
+                        new_line.append([id_val, int(val2)])
+                    else:
+                        new_line.append([id_val, val2])
+
+                # Add the finished line to the list
+                gcode.append(new_line)
+
+        return gcode
+
+    def __iter__(self):
+        """ Initialization of the the class variables which are responsible
+        for the iteration
+
+        Returns:
+        self(GCodeProcessor): Iterator
+        """
+
+        self.elementary_operations = []
+        self.index_operation = 0
+        self.index_gcode = 0
+        self.path = []
+        return self
+
+    def __next__(self):
+        """Switches between running elementary operation and reading
+        in commands from the G-code to create elementary operations.
+        Every line of the G-code causes the built
+        of new elementary operations"""
+
+        # Runs elementary operations
+        if self.index_operation < len(self.elementary_operations):
+            i = self.index_operation
+            self.index_operation += 1
+            self.elementary_operations[i]()
+
+        # Reads the G-Code command to create elementary operations
+        elif self.index_gcode < len(self.gcode):
+            self.elementary_operations = []
+            self.index_operation = 0
+
+            i = self.index_gcode
+            cmd_type = self.gcode[i][0][0]
+            cmd_int = self.gcode[i][0][1]
+
+            self.last_point = np.array(self.new_point)
+            self.last_or = np.array(self.new_or)
+
+            if cmd_type == "M":
+                self.create_elementary_operations(
+                    m_int=cmd_int)
+
+            if cmd_type == "T":
```

Comment:
should this not also be an elif?

Reply:
Changed to:  
`self.create_elementary_operations(cmd_type,cmd_int)`